source/grid/grid_tools_cache.cc

```diff
@@ -130,10 +130,10 @@ namespace GridTools
         std::vector<std::pair<
           BoundingBox<spacedim>,
           typename Triangulation<dim, spacedim>::active_cell_iterator>>
-                     boxes(tria->n_active_cells());
-        unsigned int i = 0;
+          boxes;
+        boxes.reserve(tria->n_active_cells());
```

Comment:
This looks like the same problem as #15041 - can you apply the same fix?

Reply:
Actually, I'm wrong here - while this might not be the right choice for `p::s::T` this is consistent with the present implementation.